Add docker stack and complement existing

joao10lima opened the pull request:
As described in the issue #23, using as example the GitHub starters CI, add to the Pipelinit the following stacks:
- Docker stack
  - Lint stage
  - Build stage
- Java stack
  - Build stage with Gradle 
- Complement python stack
  - Django and PyTest
  - Black as formatter and flake8 adjusted
- Complement JavaScript stack
  - Add a stage test

To test:
- Check if the unit tests pass 
- Try to generate the Pipelinit CI accordingly to each stack using the sample projects https://github.com/pipelinit/pipelinit-cli/tree/feat/github-starter-stacks#examples


![image](https://user-images.githubusercontent.com/30262244/132575583-6a3be9da-e669-454c-a59b-a4a628c9981f.png)
![image](https://user-images.githubusercontent.com/30262244/132575640-dff5c6c8-138c-4a43-98b7-ff01227cc83b.png)
![image](https://user-images.githubusercontent.com/30262244/132575696-0001bf29-0026-451c-a160-2702a5f60438.png)
![image](https://user-images.githubusercontent.com/30262244/132575864-eab78019-53e0-481f-984d-9ad43e1b24d5.png)

oesgalha commented on core/plugins/stack/python/django.ts:
```diff
@@ -0,0 +1,9 @@
+
+export const introspect: IntrospectFn<boolean> = async (context) => {
+  // Search for project manage.py to define as a Django project
+  for await (const _file of context.files.each("**/manage.py")) {
+    return true;
+  }
+  return false;
```
Use the `includes` helper to do this: 
```suggestion
  return await context.files.includes("**/manage.py");
```

oesgalha commented on core/templates/github/javascript/test.yaml:
```diff
@@ -14,4 +14,22 @@ jobs:
+        with:
+          node-version: '<%= it.runtime.version %>'
+          cache: '<%= it.packageManager.name %>'
+      - run: <%= it.packageManager.name %> ci
```
This should be `npm ci` for npm or `yarn` for yarn.

oesgalha commented on core/templates/github/python/lint.yaml:
```diff
@@ -9,5 +12,7 @@ jobs:
+
+      - run: python -m pip install pip flake8 black
+      - run: black . --check
+      - run: flake8 --ignore E203,E501,W503 .
```
Can you add a comment here explaining why those should be ignored?

oesgalha reviewed (changes requested):
Thank you!

I've made some suggestions below, can you take a look?

---

In the README file, Increase the rowspan from the cells with "Coming soon" from 8 to 13. This will make the last columns render with all the rows.

joao10lima commented:
> Thank you!
> 
> I've made some suggestions below, can you take a look?
> 
> In the README file, Increase the rowspan from the cells with "Coming soon" from 8 to 13. This will make the last columns render with all the rows.

Thanks for the comments, I've changed accordingly and updated the README example section text
